From 3ba1d250be96b9bce21e2688f03ddb6179d8c8a7 Mon Sep 17 00:00:00 2001
From: Catalin Horghidan <catalin.horghidan@nxp.com>
Date: Mon, 11 Feb 2019 17:50:10 +0200
Subject: [PATCH 630/706] felix: update skb inet csum after xtr header pull

Signed-off-by: Catalin Horghidan <catalin.horghidan@nxp.com>
(cherry picked from commit c2cf78ba5ed6ceff250cd2aeb951238f04aa06bf)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index 9fed70e3c77b..fdb4be2943fa 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -216,6 +216,7 @@ static rx_handler_result_t felix_frm_ext_handler(struct sk_buff **pskb)
 	struct net_device *ndev = (*pskb)->dev;
 	struct sk_buff *skb = *pskb;
 	struct ocelot_port *port;
+	void *start = skb->data;
 	struct ocelot *ocelot;
 	u64 *efh;
 	u32 p;
@@ -247,7 +248,9 @@ static rx_handler_result_t felix_frm_ext_handler(struct sk_buff **pskb)
 	skb_reset_transport_header(skb);
 	skb_reset_network_header(skb);
 	skb->pkt_type = PACKET_HOST;
-	skb->ip_summed = CHECKSUM_NONE;
+
+	/* update inet csum if already computed */
+	skb_postpull_rcsum(skb, start, XFH_LONG_PREFIX_LEN);
 
 	/* frame for CPU */
 	if (p == FELIX_EXT_CPU_PORT_ID)
-- 
2.17.1

